app/models/staff_response.rb

```diff
@@ -5,6 +5,8 @@ class StaffResponse
   attr_accessor :visit, :user
 
   before_validation :check_slot_available
+  before_validation :check_lead_visitor_not_banned
+  before_validation :check_lead_visitor_on_list
 
   validate :validate_visit_is_processable
   validate :visit_or_rejection_validity
```

Comment:
Is not related to book to nomis as we split into what the card for this PR. Behaviour seems right, is just the logic on that method might be possible to simplify it since we don't check for at least one adult with this PR.